### ISSUE TITLE
Updated solution 

### ISSUE THREAD

**Issue by dberlow** [twardoch/test-fonts#4]
If it applies, the illustration below by @Arrowtype, from discussion here:
twitter.com/arrowtype/status/1312023807939444736?s=21

That looks like the far right below:
![7FE25D05-5DFD-4617-B967-CD468AAF2822](https://user-images.githubusercontent.com/1956494/98113436-a8f99a00-1e71-11eb-96bd-98b36a2586e6.png)

**Comment by twardoch** [comment]
Yes, I think the 2nd-right solution is overly complicated, but the rightmost solution seems OK. 

The "problem" occurs when contours intersect in such a way that two one contour is inside another in two places and goes outside in the middle. 

Simply slicing (chopping) that contour in half should help, and it's an easy to comprehend and implement solution, even automatable.
